plugins/modules/cli_restore.py

```diff
@@ -108,12 +108,16 @@ def main():
 
     result = {"changed": False}
     connection = Connection(module._socket_path)
-    running = connection.restore(
-        filename=module.params["filename"],
-        path=module.params["path"],
-    )
-    result["__restore__"] = running
+    try:
+        running = connection.restore(
+            filename=module.params["filename"],
+            path=module.params["path"],
+        )
+    except ConnectionError as exc:
```

Comment:
This is incorrectly going to mask other issues triggering ConnectionError in platform cliconf restore(). And will fail with the unimplemented error even if it's implemented.

Reply:
yeah I can check for `method not found ` error and then catch that exception here

Reply:
Or, just use `getattr(connection, "restore")`?

Reply:
if it's None, then raise a error with your custom error message....which should include `ansible_network_os` also.

Reply:
Else:
```
running = restore_method(
            filename=module.params["filename"],
            path=module.params["path"],
        )
```

Reply:
this won't work at least directly for the bounded method ( self.__rpc__=<bound method Connection.__rpc__ of <ansible.module_utils.connection.Connection object>>)
I think we should be fine  without giving network_os.

Reply:
I'm unable to try this locally, but can you share which part doesn't work? Calling the method or checking if the method is defined? If it's the former, then we can use:
```
if hasattr(connection, "restore", None):
    running = connection.restore(
            filename=module.params["filename"],
            path=module.params["path"],
        )
else:
   raise ConnectionError("<error message>")
```

I recommend specifying the network_os value. Since we already have the connection object, we should be able to grab it. If not, then we can skip.

Reply:
Okay - it might always return True, because of https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/connection.py#L132

Wondering if there's a better way to do this than inspecting the exception message.

Reply:
One option can be to add a `restore()` method in cliconfbase which by default raises ConnectionError("this method does not seem to be implemented for <network_os>"). So, if it's not implemented in child cliconfs, this is raised, else it picks up the child implementation.

Reply:
> Wondering if there's a better way to do this than inspecting the exception message.

JSON-RPC has [standard error codes](https://www.jsonrpc.org/specification#error_object) that [we implement](https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/jsonrpc.py#L102). `if exc.code == -32601:` seems like a perfectly defensible (if opaque, so please comment what's happening) way of determining that a method doesn't exist.

Reply:
@Qalthos yes by default we do get "Method not found",  and  with these changes we are providing custom message "This platform does not support restore_plugin.Please report an issue against this platform's cliconf plugin."
In addition to this should we also return "Method not found",  error message ?